### PR TITLE
Started upgrading the plugin to CakePHP 4.0

### PR DESCRIPTION
**Pull request by dereuromark**
I can open the PR for feedback giving.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,12 +15,12 @@
...
     "require": {
         "php": ">=7.0",
-        "cakephp/cakephp": "^3.5"
+        "cakephp/cakephp": "^4.0"
     },
     "require-dev": {
-        "dereuromark/cakephp-ide-helper": "@stable",
-        "phpunit/phpunit": "^5.7.14|^6.0",
-        "cakephp/cakephp-codesniffer": "^3.0"
+        "phpunit/phpunit": "~8.5.0",
```
Comment (dereuromark) [inline composer.json]: I would keep `^` here - as it is unlikely the require-dev dependency here will create issues in minors.

**Comment by dereuromark** [comment]
https://github.com/dereuromark/cakephp-service-layer/commit/65b3a802cad2fdac690fb995629d8a9a85275a99
=> you can pull this in as diff patch

But then 1 failure remains

> 1) Burzum\Cake\Service\PaginationServiceTest::testPaginate
Failed asserting that null is of type "array".

**Comment by burzum** [comment]
@dereuromark I'm aware of the issue but haven't had time and mood to find a good solution. The code passed the request object as reference around but when you now manipulate it, you get a clone because it's immutable. So the whole concept doesn't work anymore. I've already tried passing the object back, but this would require you to do two instead of one call and you MUST do the 2nd one after the paginate() call.

**Comment by dereuromark** [comment]
After applying my diff/changes you can probably start to release a beta here, so we can already use it - and also mention it as plugin in https://github.com/FriendsOfCake/awesome-cakephp
The small pagination issue so far no one of us is using anyway, most are using the locator topic.

**Comment by burzum** [comment]
@dereuromark I would prefer to agree on a solution to the pagination before doing a release. I have some local changes to it already, but as I said this would require the user of the plugin to do this:

```php
        $result = $service->paginate($articles);
        $request = $service->addPagingParamToRequest($request);
```

instead of just calling the first line as before.

We also need to get CI pass again. I just removed the old php versions from CI and set it to 7.3 and 7.4.

**Pull request merged**